stdlib/public/SwiftShims/RefCount.h

```diff
@@ -18,7 +18,7 @@
 // This definition is a placeholder for importing into Swift.
 // It provides size and alignment but cannot be manipulated safely there.
 typedef struct {
-  __swift_uintptr_t refCounts SWIFT_ATTRIBUTE_UNAVAILABLE;
```

Comment:
So, in C-mode this attribute is meaningless which is why we didn't see the error before now. If we want to make sure that in runtime mode we don't use this type, I'll need to create a separate and more involved fix. Let me know if that needs to be done. 

Another "fix" could be to only add `SWIFT_ATTRIBUTE_UNAVAILABLE` when `ifndef __swift__ `.

Reply:
> So, in C-mode this attribute is meaningless which is why we didn't see the error before now.

Which error?

Reply:
When we have a reference type, for example an array, we try to add an implicit copy constructor in C++ mode. When we try to do this, we get an error that `refCounts` is unreachable: 
```
swift-source/build/Ninja-ReleaseAssert/swift-macosx-arm64/lib/swift/shims/RefCount.h:20:9: error: 'refCounts' is unavailable
typedef struct {
        ^
swift-source/build/Ninja-ReleaseAssert/swift-macosx-arm64/lib/swift/shims/HeapObject.h:45:8: note: in implicit copy constructor for 'InlineRefCountsPlaceholder' first required here
struct HeapObject {
       ^
swift-source/build/Ninja-ReleaseAssert/swift-macosx-arm64/lib/swift/shims/GlobalObjects.h:38:8: note: in implicit copy constructor for 'HeapObject' first required here
struct _SwiftEmptyArrayStorage {
       ^
<unknown>:0: note: in implicit copy constructor for '_SwiftEmptyArrayStorage' first required here
swift-source/build/Ninja-ReleaseAssert/swift-macosx-arm64/lib/swift/shims/RefCount.h:21:21: note: 'refCounts' has been explicitly marked unavailable here
  __swift_uintptr_t refCounts SWIFT_ATTRIBUTE_UNAVAILABLE;
```